File: CarHomeWidget/src/main/java/com/anod/car/home/skin/CardsProperties.java
package com.anod.car.home.skin;

import com.anod.car.home.R;
import com.anod.car.home.skin.icon.BackgroundProcessor;
import com.anod.car.home.skin.icon.CardsBackgroundProcessor;

public class CardsProperties extends BaseProperties {

    @Override
    public int getInCarButtonExitRes() {
        return R.drawable.ic_incar_exit_gray;
    }

    @Override
    public int getInCarButtonEnterRes() {
        return R.drawable.ic_incar_enter_gray;
    }

    @Override
    public int getSetShortcutRes() {
        return R.drawable.ic_add_shortcut_holo;
    }

    @Override
    public int getLayout(int number) {
        if (number == 4) {
            return R.layout.sk_material_4;
        }
        if (number == 8) {
            return R.layout.sk_material_8;
        }
        return R.layout.sk_material_6;
    }

    @Override
    public int getSettingsButtonRes() {
        return R.drawable.ic_settings_grey600_36dp;
    }

    @Override
    public int getRowLayout() {
        return R.layout.sk_material_row;
    }

    @Override
    public int getIconPaddingRes() {
        return 0;
    }

    @Override
    public boolean hasWidgetButton1() {
        return false;
    }

    @Override
    public BackgroundProcessor getBackgroundProcessor() {
        return new CardsBackgroundProcessor();
    }
}

File: app/src/main/java/com/anod/car/home/skin/CardsProperties.kt
package com.anod.car.home.skin

import com.anod.car.home.R
import com.anod.car.home.skin.icon.CardsBackgroundProcessor
import info.anodsplace.carwidget.content.IconBackgroundProcessor

class CardsProperties : BaseProperties() {

    override val inCarButtonExitRes: Int
        get() = R.drawable.ic_incar_exit_gray

    override val inCarButtonEnterRes: Int
        get() = R.drawable.ic_incar_enter_gray

    override val setShortcutRes: Int
        get() = R.drawable.ic_add_shortcut_holo

    override val settingsButtonRes: Int
        get() = R.drawable.ic_settings_grey600_36dp

    override val rowLayout: Int
        get() = R.layout.sk_material_row

    override val iconPaddingRes: Int
        get() = 0

    override val backgroundProcessor: IconBackgroundProcessor
        get() = CardsBackgroundProcessor()

    override fun getLayout(number: Int): Int {
        return when (number) {
            4 -> R.layout.sk_material_4
            8 -> R.layout.sk_material_8
            10 -> R.layout.sk_material_10
            else -> R.layout.sk_material_6
        }
    }

    override fun hasWidgetButton1(): Boolean {
        return false
    }
}